org-code-javabuilder/theater/src/main/java/org/code/theater/Stage.java

```diff
@@ -362,8 +366,14 @@ public void play() {
       this.gifWriter.close();
       this.audioWriter.writeToAudioStreamAndClose();
 
-      this.outputAdapter.sendMessage(ImageEncoder.encodeStreamToMessage(this.imageOutputStream));
-      this.outputAdapter.sendMessage(SoundEncoder.encodeStreamToMessage(this.audioOutputStream));
+      // If we are in local mode, write output data directly to output adapter
+      if (GlobalProtocol.getInstance().getDashboardHostname().contains("localhost")) {
```

Comment:
Rather than checking for the string "localhost" here, we should create an interface that is implemented differently for local vs aws instances of javabuilder. That way Stage.java doesn't need to know if the dashboard hostname contains localhost and can invoke the interface directly.

Reply:
What kind of interface are you thinking of? I thought about doing something on the file writer but we need to encode the image/audio and it doesn't make much sense to move that over to `lib`